packages/formatron/src/dataType.ts

```diff
@@ -23,8 +22,28 @@ export abstract class DataType<
    */
   abstract isOfType(value: unknown): value is T;
 
-  get isRequired(): boolean {
-    return this.field.options?.required ?? false;
+  isRequired(model?: AnyModel): boolean {
+    const required = this.field.options?.required;
+    switch (typeof required) {
+      case 'object': {
+        if (!model) {
+          console.warn(
+            `Model is needed to validate if field '${this.field.name}' is required`
+          );
+          return false;
+        }
+        const path = required.path.map((item) =>
+          item === 'relations' ? 'relationModels' : item
+        );
+        const value = getIn(model, path);
+        const thisCase = required.cases.find(({pattern}) => pattern === value);
+        return thisCase?.result ?? false;
+      }
+      case 'boolean':
+        return required;
+      default:
```

Comment:
Could have a `case 'undefined':` here that just returns `false`, then the default case either throws an error, or if we want to be loose, logs a warning then returns false.  Or something similar so that when the value is something dumb like `2`, we don't hide that fact.

Reply:
Good call.